base58/Cargo.toml

```diff
@@ -22,7 +22,7 @@ rustdoc-args = ["--cfg", "docsrs"]
 
 [dependencies]
 hashes = { package = "bitcoin_hashes", version = "0.14.0", default-features = false, features = ["alloc"] }
-internals = { package = "bitcoin-internals", version = "0.3.0" }
+internals = { package = "bitcoin-internals", version = "0.3.0", features = ["alloc"] }
```

Comment:
I am somewhat confused by this commit. 

Are we adding this just to indicate that we need an allocator, but adding the feature is necessary for the code to compile and work correctly?

Reply:
We already have `alloc` on in `hashes`, as you can see in the above line. So there is already no way to compile base58ck without an allocator (maybe there should be, using some sort of `SmallVec` magic, but that's a future job).

*However*, if you compile base58 by itself, it will pull in `internals` without the `alloc` feature, which disables several impls in the `impl_parse` macro and also disables most of the internals of the `InputString` type. So we have an allocator but we're not making full use of it.